MediaContentSynchronizationCatalog/Model/Synchronizer/Product.php

```diff
@@ -83,19 +76,21 @@ public function __construct(
      */
     public function execute(): void
     {
-        foreach ($this->repository->getList($this->searchCriteriaBuilder->create())->getItems() as $item) {
-            foreach ($this->fields as $field) {
-                $contentIdentity = $this->contentIdentityFactory->create(
-                    [
-                        self::TYPE => self::CONTENT_TYPE,
-                        self::FIELD => $field,
-                        self::ENTITY_ID => $item->getId()
-                    ]
-                );
-                $this->updateContentAssetLinks->execute(
-                    $contentIdentity,
-                    implode(PHP_EOL, $this->getEntityContents->execute($contentIdentity))
-                );
+        foreach ($this->selectBatches->execute(self::PRODUCT_TABLE, [self::PRODUCT_TABLE_ENTITY_ID]) as $batch) {
```

Comment:
entity_id field names may need to be retrieved dynamically (row_id is introduced in commerce staging). Needs to be checked.